### PR TITLE
EL-3345 - improve localization experience

### PR DESCRIPTION
**Pull request by TaraTodd123**
#### Checklist

* [x] The contents of this PR are mine to submit. No third party code or other material is being committed.
* [x] New third party dependencies (if any) are linked via `package.json`. Third party dependencies are licensed under one of the following: Apache, MIT, BSD, Mozilla Public License, Eclipse Public License, or Oracle Binary Code License.
* [x] The code in this PR does not contain export-restricted encryption algorithms.

#### Ticket / Issue

https://portal.digitalsafe.net/browse/EL-3345

#### Description of Proposed Changes

- Removing all uses of the i18n Angular functionality. Instead all strings that previous used this are now defined using inputs instead.
- Because some aria labels need a function to be called to produce a valuable label, I have had to rework a couple of components a bit to allow them to use OnPush change detection, otherwise the aria label functions were getting called many time every time change detection runs.

#### Documentation CI URL

https://pages.github.houston.softwaregrp.net/sepg-docs-qa/UXAspects_CI_UXAspects-2_EL-3345-improve-localization-experience

### CODE REVIEW

**Review by donna-marijne — CHANGES_REQUESTED** [review]
There seems to be a number of changes here that are unrelated to i18n. Could you separate those out and create a ticket so that they can receive proper QA in a future sprint?

**Review by donna-marijne — CHANGES_REQUESTED** [review]
Thanks Tara, tested a few of these and all looks well. Just one leftover change that I'm a bit concerned about.

**Comment by TaraTodd123** [comment]
What change is it? I don't know if there was a comment for it or not?

File: src/components/color-picker/color-picker.component.ts [hunk 1]
```diff
@@ -17,7 +17,8 @@ let uniqueId = 0;
 @Component({
     selector: 'ux-color-picker',
     exportAs: 'ux-color-picker',
-    templateUrl: 'color-picker.component.html'
+    templateUrl: 'color-picker.component.html',
+    changeDetection: ChangeDetectionStrategy.OnPush
```
Comment (donna-marijne) [inline src/components/color-picker/color-picker.component.ts]: Looks like I forgot to click the button - this one.

Comment (TaraTodd123) [reply src/components/color-picker/color-picker.component.ts]: removed and seems to work the same

**Review by donna-marijne — APPROVED** [review]
All good.

**Pull request merged**